lib/reservation.rb

```diff
@@ -0,0 +1,47 @@
+require "date"
+require "time"
+
+module Hotel
+  class Reservation
+    attr_reader :booking_ref
+    attr_accessor :booking_ref, :number_of_rooms, :first_room_number, :check_in, :check_out, :total_cost, :room_blocks
```

Comment:
You probably should make all of these `reader`. Also, you don't need to double up-- `booking_ref` is in both `reader` and `accessor` and you should probably pick one